src/content/pt/2019/compression.md

```diff
@@ -0,0 +1,364 @@
+
+No exemplo seguinte, o cliente comunica que tem suporte para compressão Gzip, Brotli e Deflate. O servidor decidiu retornar uma resposta comprimida com GZip que contém um documento `text/html`.
+
+```
+    > GET / HTTP/1.1
+    > Host: httparchive.org
+    > Accept-Encoding: gzip, deflate, br
+
+    < HTTP/1.1 200
+    < Content-type: text/html; charset=utf-8
+    < Content-encoding: gzip
+```
+
+O HTTP Archive contém leituras de 5.3 milhões de sites e cada site carregou pelo menos 1 recurso de texto comprimido na sua página de início. Adicionalmente, 81% dos sites reportaram ter recursos comprimidos no seu domínio primário.
+
+## Algoritmos de compressão
+
+A organização IANA mantém uma [lista de codificação de conteúdos HTTP válidos](https://www.iana.org/assignments/http-parameters/http-parameters.xml#content-coding) que podem ser usados com os cabeçalhos `Accept-Encoding` e `Content-Encoding`. Alguns destes incluem `gzip`, `deflate`, `br` (Brotli) entre outros. Uma breve descrição destes algoritmos é apresentada de seguida:
+
+* [Gzip](https://tools.ietf.org/html/rfc1952) usa as técnicas de compressão [LZ77](https://en.wikipedia.org/wiki/LZ77_and_LZ78#LZ77) e [Huffman coding](https://pt.wikipedia.org/wiki/Codifica%C3%A7%C3%A3o_de_Huffman) e é mais antigo que a própria web. Foi originalmente desenvolvido para o programa `gzip` do Unix em 1992. Uma implementação para servir conteúdos na web existe desde o HTTP/1.1 e a grande maioria dos browsers e clientes suportam este algoritmo de compressão.
+* [Deflate](https://tools.ietf.org/html/rfc1951) usa o mesmo algoritmo que o Gzip mas com um recipiente diferente. Não foi tão adoptado para a web devido a [problemas de compatibilidade](https://en.wikipedia.org/wiki/HTTP_compression#Problems_preventing_the_use_of_HTTP_compression) com alguns servidores e browsers.
+* [Brotli](https://tools.ietf.org/html/rfc7932) é um novo algoritmo de compressão [criado pela Google](https://github.com/google/brotli). Combina a variante moderna da técnica de compressão LZ77 e Huffman e uma modelação de contexto de segunda ordem. A compressão via Brotli é computacionalmente mais exigente que o Gzip mas o algoritmo é capaz de reduzir o tamanho dos ficheiros em [15-25%](https://cran.r-project.org/web/packages/brotli/vignettes/brotli-2015-09-22.pdf) em relação à compressão Gzip. O Brotli foi inicialmente usado para compressão de conteúdos web em 2015 e é [suportado em todos os browsers modernos](https://caniuse.com/#feat=brotli).
+
+Aproximadamente 38% das respostas HTTP são servidas com compressão baseada em texto. Poderá parecer uma estatística surpreendente mas deve levar em conta que é baseado em todos os pedidos HTTP no dataset. Algum conteúdo, como imagens, não irá beneficiar destes algoritmos de compressão. A tabela abaixo resume a percentagem de pedidos servidos para cada tipo de codificação de contéudo.
```

Comment:
Changes:

* `Poderá parecer uma estatística surpreendente mas deve levar em conta que é baseado em...` -> `Pode parecer uma estatística surpreendente mas deve-se levar em conta que é baseada em...`
* `percentagem` -> `porcentagem`.

The same occurs on the table below